dll/win32/msgina/shutdown.c

```diff
@@ -324,6 +483,173 @@ LoadShutdownSelState(VOID)
     return dwValue;
 }
 
+static INT_PTR
+CALLBACK
+HotButtonSubclass(
+    HWND hButton,
+    UINT uMsg,
+    WPARAM wParam,
+    LPARAM lParam)
+{
+    PSHUTDOWN_DLG_CONTEXT pContext;
+    pContext = (PSHUTDOWN_DLG_CONTEXT)GetWindowLongPtrW(hButton, GWLP_USERDATA);
+
+    int buttonID = GetDlgCtrlID(hButton);
+
+    switch (uMsg)
+    {
+        case WM_MOUSEMOVE:
+        {
+            POINT pt;
+
+            if (GetCapture() != hButton)
+            {
+                SetCapture(hButton);
+                if (buttonID == IDC_BUTTON_SHUTDOWN)
+                {
+                    pContext->bIsButtonHot[0] = TRUE;
+                }
+                else if (buttonID == IDC_BUTTON_REBOOT)
+                {
+                    pContext->bIsButtonHot[1] = TRUE;
+                }
+                else if (buttonID == IDC_BUTTON_SLEEP)
+                {
+                    pContext->bIsButtonHot[2] = TRUE;
+                }
+                else if (buttonID == IDC_BUTTON_HIBERNATE)
+                {
+                    pContext->bIsButtonHot[3] = TRUE;
+                }
+                SetCursor(LoadCursorW(NULL, MAKEINTRESOURCEW(IDC_HAND)));
+            }
+
+            pt = (POINT){GET_X_LPARAM(lParam), GET_Y_LPARAM(lParam)};
+            ClientToScreen(hButton, &pt);
+            if(WindowFromPoint(pt) != hButton)
```

Comment:
Substitute the return value into a variable.
```suggestion
            if (hwndTarget != hButton)
```